### PR TITLE
New Testutil_mock.mli

### PR DESCRIPTION
**Pull request by aryx**
Factorized code from Unit_LS.ml and Test_login_subcommand.ml

test plan:
make core
make core-test

### CODE REVIEW

File: js/node_shared/unix.js [hunk 1]
```diff
@@ -265,6 +265,12 @@ function unix_fork(vunit) {
   return;
 }
 
+//Provides: unix_putenv
+function unix_putenv(key, val) {
+  //TODO: nodejs equivalent?
```
Comment (ajbt200128) [inline js/node_shared/unix.js]: https://stackoverflow.com/questions/10829433/how-can-i-set-an-environment-variable-in-node-js

**Review by ajbt200128 — APPROVED** [review]

File: js/node_shared/unix.js (hunk 1 above)
Comment (ajbt200128) [inline js/node_shared/unix.js]:
```suggestion
  process.env[key] = value
```

**Comment by aryx** [comment]
@ajbt200128 @tpetr any idea why I get those new JS failures after my PR:
```
OK]          Language Server (unit) > Processed Run                               3   Test only git dirty with dirty files....
2023-11-20T10:08:25.0666735Z   [OK]          Language Server (unit) > Processed Run                               4   Test nosem....
2023-11-20T10:08:25.0693739Z Language Server (unit)>CI Tests>Test session cache
2023-11-20T10:08:25.0698769Z ReferenceError: value is not defined
2023-11-20T10:08:25.0769859Z ReferenceError: value is not defined
2023-11-20T10:08:25.0770586Z     at abO (/__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:1780:25)
2023-11-20T10:08:25.0771441Z     at /__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:76857:12
2023-11-20T10:08:25.0772257Z     at n (/__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:1873:43)
2023-11-20T10:08:25.0773083Z     at boW (/__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:77184:3)
2023-11-20T10:08:25.0773908Z     at /__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:77254:45
2023-11-20T10:08:25.0774716Z     at n (/__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:1873:43)
2023-11-20T10:08:25.0775533Z     at e (/__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:44373:26)
2023-11-20T10:08:25.0776336Z     at n (/__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:1873:43)
2023-11-20T10:08:25.0777148Z     at gsF (/__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:44161:3)
2023-11-20T10:08:25.0777958Z     at l (/__w/semgrep/semgrep/_build/default/js/tests/test_jsoo.bc.js:1874:45)
2023-11-20T10:08:25.0813968Z   [FAIL]        Language Server (unit) > CI Tests                                    0   Test session cache....
2023-11-20T10:08:25.0851965Z   [OK]          Language Server (unit) > Lib EV tests                                0   Test LS with libev....
2023-11-20T10:08:25.0852684Z 
2023-11-20T10:08:25.0853526Z ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
2023-11-20T10:08:25.0854980Z │ [FAIL]        Language Server (unit) > CI Tests                                    0   Test session cache.... │
2023-11-20T10:08:25.0856800Z └───────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
2023-11-20T10:08:25.0860053Z 
2023-11-20T10:08:25.0860285Z ReferenceError: value is not defined
2023-11-20T10:08:25.0861219Z [exception] ReferenceError: value is not defined
2023-11-20T10:08:25.0865359Z Logs saved to `/__w/semgrep/semgrep/js/tests/_build/_tests/semgrep-js/Language Server U+0028unitU+0029 U+003E CI Tests.000.output'.
```

**Comment by aryx** [comment]
I can't even reproduce locally :( make test is working fine under js/ on my machine (after I've commented the build of julia which fails with a segfault on my machine)

**Comment by aryx** [comment]
grrr
```
┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
2023-11-20T10:44:17.6197678Z │ [FAIL]        OSemgrep Fetch > fetch tests                                         0   fetch ocaml rules 1.... │
2023-11-20T10:44:17.6199192Z └────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
2023-11-20T10:44:17.6200504Z [36.38][[32mDEBUG[0m]: trying to download from https://semgrep.dev/c/p/ocaml
2023-11-20T10:44:17.6201637Z [36.38][[32mDEBUG[0m]: GET on https://semgrep.dev/c/p/ocaml
2023-11-20T10:44:17.6203031Z [36.38][[32mDEBUG[0m]: finished downloading from https://semgrep.dev/c/p/ocaml
2023-11-20T10:44:17.6204594Z [failure] TLS to non-TCP currently unsupported: host=semgrep.dev endp=(Unknown "name resolution failed")
2023-11-20T10:44:17.6206588Z           Raised at Http_helpers.Make.get_async.(fun) in file "libs/networking/http_helpers/http_helpers.ml", line 51, characters 4-722
2023-11-20T10:44:17.6208754Z           Re-raised at Rule_fetching.fetch_content_from_url_async.(fun) in file "src/osemgrep/networking/Rule_fetching.ml", line 119, characters 4-294
2023-11-20T10:44:17.6211004Z           Re-raised at Rule_fetching.rules_from_dashdash_config_async.(fun) in file "src/osemgrep/networking/Rule_fetching.ml", line 444, characters 6-474
2023-11-20T10:44:17.6212914Z           Re-raised at Lwt.Miscellaneous.poll in file "src/core/lwt.ml", line 3123, characters 20-29
2023-11-20T10:44:17.6215886Z           Called from Lwt_main.run.run_loop in file "src/unix/lwt_main.ml", line 27, characters 10-20
2023-11-20T10:44:17.6217332Z           Called from Lwt_main.run in file "src/unix/lwt_main.ml", line 106, characters 8-13
2023-11-20T10:44:17.6218683Z           Re-raised at Lwt_main.run in file "src/unix/lwt_main.ml", line 112, characters 4-13
2023-11-20T10:44:17.6220516Z           Called from Unit_Fetching.real_fetch_tests.fetch_ocaml_rules in file "src/osemgrep/networking/Unit_Fetching.ml", line 28, characters 6-150
2023-11-20T10:44:17.6222637Z           Called from Alcotest_engine__Core.Make.protect_test.(fun) in file "src/alcotest-engine/core.ml", line 181, characters 17-23
2023-11-20T10:44:17.6224576Z           Called from Alcotest_engine__Monad.Identity.catch in file "src/alcotest-engine/monad.ml", line 24, characters 31-35
2023-11-20T10:44:17.6225694Z           
```

**Pull request merged**